### ISSUE TITLE
RNN base class doesn't support dropout 

### ISSUE THREAD

**Issue by mpariente** [keras-team/keras#8593]
I'm using the latest version of Keras and Tensorflow and when I run this little code, I get an TypeError 
```
from keras import backend as K
from keras.layers import Input
from keras.layers.recurrent import RNN, LSTMCell, SimpleRNNCell, GRUCell
from keras.models import Model
from keras.optimizers import Nadam

cells = [LSTMCell(8, dropout=0.1, recurrent_dropout=0.1),
         LSTMCell(8, dropout=0.1, recurrent_dropout=0.1)]
x = Input((None, 8))
layer = RNN(cells, return_sequences=True)
y = layer(x)

model = Model(inputs = x, outputs = y)
model.compile(loss='mean_squared_error', optimizer='Nadam')
```
- Error occurs using both Tensorflow and Theano as backend (didn't try CNTK though)
- Error occurs using both implementation mode (1 and 2)
- Error occurs using `GRUCell` as well
- Error doesn't occur if I don't use `dropout` and `recurrent_dropout`.
- Error doesn't occur if I use `SimpleRNNCell` 

Is anyone able to reproduce this error? 
I can give the Traceback if you want but it seems that `dp_mask` and `recurrent_dp_mask` are not properly initialized
The problem seems to be between [here](https://github.com/fchollet/keras/blob/master/keras/layers/recurrent.py#L1797) and [here](https://github.com/fchollet/keras/blob/master/keras/layers/recurrent.py#L1806).
